### PR TITLE
fix: MinorVersionCompatibility test

### PR DESCRIPTION
**Pull request by smuu**
## Overview

This fixes the port forwarding issues nina faces

cc: @ninabarbakadze 

Closes https://github.com/celestiaorg/celestia-app/issues/3187

### CODE REVIEW

File: test/e2e/main.go [hunk 1]
```diff
@@ -24,8 +24,7 @@ func main() {
 	logger := log.New(os.Stdout, "test-e2e", log.LstdFlags)
 
 	tests := []Test{
-		// FIXME the following test is currently failing.
-		// {"MinorVersionCompatibility", MinorVersionCompatibility},
```
Comment (rootulp) [inline test/e2e/main.go:28]: you're a beast @smuu thanks so much!

**Review by rootulp — APPROVED** [review]
Thanks! I think you can update the PR description to state this PR closes https://github.com/celestiaorg/celestia-app/issues/3187

Also, I think golangci-lint is complaining so we prob need to fix that before merging but we really appreciate this fix

**Review by ninabarbakadze — DISMISSED** [review]
thank you so much! I can resolve the lint issues for you if you like

**Comment by smuu** [comment]
Adding some context/explanation:

Using port-forwarding in Kubernetes is error-prone, and I try to avoid it whenever possible.

In this case, the problem is:
- We start an Instance
- We create a port-forward
- We replace the running Pod with another Pod with another version without closing the port-forward
- We try to use the still existing port-forward where the target changed and therefore fails

The workaround I propose does not use port-forwarding. Instead, it executed a shell command on another Instance (Executer) to fetch the height from the RPC.
I understand this is not optimal, as using the RPC client inside app makes more sense than shell scripts. If more resources are available, we will see if we can solve it in knuu.

I hope this workaround is sufficient enough for the time beeing.

**Comment by smuu** [comment]
> thank you so much! I can resolve the lint issues for you if you like

That would be great!

**Review by ninabarbakadze — COMMENTED** [review]
everything should be green now @smuu @rootulp

**Review by rootulp — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**